tests/test_providers.py

```diff
@@ -28,6 +44,7 @@ def token(mocker):
 def mock_network(mocker):
     mock = mocker.MagicMock(spec=NetworkAPI)
     mock.name = "MOCK_NETWORK"
+    mock.ecosystem.name = "MOCK_ECOSYSTEM"
```

Comment:
Does this work?
You meed to do something like:

```python
mock.ecosystem = mocker.MagicMock(spec=EcosystemAPI)
mock.ecosystem.name = "MOCK_ECOSYSTEM"
```

Reply:
Yeah, I've been getting `Mock object has no attribute 'ecosystem'` errors. Was lost because I'm a complete newbie to testing. Just added this change now.